exercise/exercise_models.py

```diff
@@ -576,7 +576,8 @@ def _check_submission_allowed(self, profile, request=None):
 
         # Support group id from post or currently selected group.
         group = None
-        group_id = request.POST.get("_aplus_group") if request else None
+        group_id = json.loads(request.POST.get('__aplus__','{}')).get('group') or request.POST.get("_aplus_group") if request else None
```

Comment:
This is a pretty long line and more importantly, it includes a lot of different things (i.e. hard to follow). In addition, now the condifition is really far on the right, so it's easy to miss. Might better to use traditional `if cond: <then>; else: group_id = None`

Further, you need to handle JSON parsin errors. For an example, see another parto f the code: https://github.com/apluslms/a-plus/blob/8c7c5a19669cdcf32efd32d2cafc617cead490f8/edit_course/operations/batch.py#L15